proto_proxy.c

```diff
@@ -325,6 +351,27 @@ void proxy_return_cb(io_pending_t *pending) {
 void proxy_finalize_cb(io_pending_t *pending) {
     io_pending_proxy_t *p = (io_pending_proxy_t *)pending;
 
+    if (p->io_type == IO_PENDING_TYPE_EXTSTORE) {
+        if (p->hdr_it) {
+            // TODO: lock once, worst case this hashes/locks twice.
+            if (p->miss) {
+                item_unlink(p->hdr_it);
+            }
+            item_remove(p->hdr_it);
+        }
+    } else {
+        // free our buffer here instead of relying on write_and_free since
+        // we can do memory accounting.
+        if (p->client_resp->buf) {
```

Comment:
Tempted to remove this and rely on the GC instead, just to reduce the amount of code a bit.

The tradeoff would be memory sticking around a little longer, but it's pretty much always cleaned up nearly immediately anyway...

Reply:
Actually this is a bug I think. the response buffer might still be used later. Will delete